adapters/rubicon/rubicon.go

```diff
@@ -649,19 +659,74 @@ func (a *RubiconAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adap
 				// set user.ext.tpid
 				if len(userExt.Eids) > 0 {
 					tpIds := make([]rubiconExtUserTpID, 0)
+
+					var segments []string
 					for _, eid := range userExt.Eids {
 						if eid.Source == "adserver.org" {
 							uids := eid.Uids
 							if len(uids) > 0 {
 								uid := uids[0]
-								if uid.Ext != nil && uid.Ext.RtiPartner == "TDID" {
-									tpIds = append(tpIds, rubiconExtUserTpID{Source: "tdid", UID: uid.ID})
+
+								if uid.Ext != nil {
+									var eidUidExt rubiconUserExtEidUidExt
+									if err := json.Unmarshal(uid.Ext, &eidUidExt); err != nil {
+										errs = append(errs, &errortypes.BadInput{
+											Message: err.Error(),
+										})
+										continue
+									}
+
+									if eidUidExt.RtiPartner == "TDID" {
+										tpIds = append(tpIds, rubiconExtUserTpID{Source: "tdid", UID: uid.ID})
+									}
+								}
+							}
+						} else if eid.Source == "liveintent.com" {
+							uids := eid.Uids
+							if len(uids) > 0 {
+								uidId := uids[0].ID
+								if uidId != "" {
+									tpIds = append(tpIds, rubiconExtUserTpID{Source: "liveintent.com", UID: uidId})
+								}
+
+								if eid.Ext != nil {
+									var eidExt rubiconUserExtEidExt
+									if err := json.Unmarshal(eid.Ext, &eidExt); err != nil {
+										errs = append(errs, &errortypes.BadInput{
+											Message: err.Error(),
+										})
+										continue
+									}
+									segments = eidExt.Segments
 								}
 							}
 						}
 					}
+
 					if len(tpIds) > 0 {
 						userExtRP.TpID = tpIds
+
+						if segments != nil {
+							userExtRPTarget := make(map[string]interface{})
+
+							if userExtRP.RP.Target != nil {
+								if err := json.Unmarshal(userExtRP.RP.Target, &userExtRPTarget); err != nil {
+									errs = append(errs, &errortypes.BadInput{
+										Message: err.Error(),
+									})
+									continue
+								}
+							}
+
+							userExtRPTarget["LIseg"] = segments
+
+							target, err = json.Marshal(&userExtRPTarget)
+							if err != nil {
+								errs = append(errs, err)
+								continue
+							}
+							userExtRP.RP.Target = target
+						}
```

Comment:
Same as above, I'd suggest extracting this into it's own function too if possible.